### PR TITLE
SALTO-2364: add service id field to empty types

### PR DESCRIPTION
**Pull request by noamhammer**
_Add service id field to empty types created upon addition of types with no instances for ducktype_

---

_Additional context for reviewer_

- [x] Need to add fieldTypeOverrides to all the relevant types

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/adapter-components/src/elements/ducktype/type_elements.ts [hunk 1]
```diff
@@ -237,26 +248,41 @@ export const generateType = ({
         ])
     )
 
-  // mark fields as hidden based on config
```
Comment (netama) [inline packages/adapter-components/src/elements/ducktype/type_elements.ts:240]: why did you remove the comment?

File: packages/adapter-components/src/elements/ducktype/type_elements.ts [hunk 2]
```diff
@@ -237,26 +248,41 @@ export const generateType = ({
...
-    isSettings: transformationConfigByType[naclName]?.isSingleton ?? false,
+    isSettings: isSingleton ?? false,
   })
 
+  if (
+    addServiceIdToType
+    && serviceIdField
+    && !Object.prototype.hasOwnProperty.call(fields, serviceIdField)
+  ) {
+    type.fields[serviceIdField] = new Field(type, serviceIdField, BuiltinTypes.UNKNOWN)
```
Comment (netama) [inline packages/adapter-components/src/elements/ducktype/type_elements.ts:269]: why do we need to add it this way? we can `fixFieldTypes` even if the field doesn't exist - there's an `if (field === undefined)` clause

File: packages/zendesk-support-adapter/src/config.ts [hunk 3]
```diff
@@ -40,6 +40,10 @@ export const FIELDS_TO_HIDE: configUtils.FieldToHideType[] = [
   { fieldName: 'updated_at', fieldType: 'string' },
 ]
 
+export const FIELD_TYPES_TO_OVERRIDES: configUtils.FieldTypeOverrideType[] = [
```
Comment (netama) [inline packages/zendesk-support-adapter/src/config.ts:43]: why does this need to be exported? and if you are defining it as a const, why not just have this single entry instead of an array? 
also, for consistency I think we should either also have a const for `{ fieldName: 'id', fieldType: 'string' }` which is defined manually below, or not have either

File: packages/zendesk-support-adapter/src/config.ts [hunk 4]
```diff
@@ -130,9 +134,9 @@ export const DEFAULT_TYPES: ZendeskApiConfig['types'] = {
     transformation: {
       sourceTypeName: 'organizations__organizations',
       fieldsToHide: FIELDS_TO_HIDE.concat({ fieldName: 'id', fieldType: 'number' }),
-      fieldTypeOverrides: [
+      fieldTypeOverrides: FIELD_TYPES_TO_OVERRIDES.concat(
```
Comment (netama) [inline packages/zendesk-support-adapter/src/config.ts:137]: continuing from above, IMO the array concatenation is an overkill here and makes this harder to read, if you want to have a const we can have one but it should probably be something like `ID_FIELD_NUMBER_OVERRIDE` and just be a single entry

File: packages/adapter-components/src/elements/ducktype/type_elements.ts [hunk 5]
```diff
@@ -168,6 +177,7 @@ export const generateType = ({
   typeNameOverrideConfig,
   isSubType = false,
   isUnknownEntry,
+  addServiceIdToType = false,
```
Comment (netama) [inline packages/adapter-components/src/elements/ducktype/type_elements.ts:180]: commented below - I might be missing something but I don't think this flag is needed if we add the field type overrides in the relevant types (which it looks like you're doing either way)

File: packages/zendesk-support-adapter/src/config.ts [hunk 6]
```diff
@@ -130,9 +132,9 @@ export const DEFAULT_TYPES: ZendeskApiConfig['types'] = {
     transformation: {
       sourceTypeName: 'organizations__organizations',
       fieldsToHide: FIELDS_TO_HIDE.concat({ fieldName: 'id', fieldType: 'number' }),
-      fieldTypeOverrides: [
+      fieldTypeOverrides: [{ fieldName: 'id', fieldType: 'number' }].concat(
```
Comment (netama) [inline packages/zendesk-support-adapter/src/config.ts:135]: why not
```suggestion
      fieldTypeOverrides: [
        { fieldName: 'id', fieldType: 'number' },
```

**Review by netama — APPROVED** [review]
LGTM with one small nit

File: packages/zendesk-support-adapter/src/config.ts (hunk 6 above)
Comment (noamhammer) [reply packages/zendesk-support-adapter/src/config.ts:135]: probably because it was missed in my find and replace
